packages/mobx-state-tree/__tests__/core/object.test.ts

```diff
@@ -605,10 +605,34 @@ if (process.env.NODE_ENV !== "production") {
         unprotect(b)
         expect(() => {
             detach(b.a)
-        }).toThrowError(/Error while converting `null` to `A`/)
+        }).toThrowError(/Error while converting `undefined` to `A`/)
         expect(() => {
             destroy(b.a)
-        }).toThrowError(/Error while converting `null` to `A`/)
+        }).toThrowError(/Error while converting `undefined` to `A`/)
+    })
+    test("it should be possible to remove a node from a parent if it is defined as type maybe ", () => {
+        const A = types.model("A", { x: 3 })
+        const B = types.model("B", { a: types.maybe(A) })
+        const b = B.create({ a: { x: 7 } })
+        unprotect(b)
+        expect(() => {
+            detach(b.a!)
+        }).not.toThrowError(/Error while converting `undefined` to `A`/)
```

Comment:
I think in this case it is better to use `not.toThrowError()` without the string argument to make sure it doesn't throw any kind of errors, not just an error with that error message

also feel free to put detach and destroy inside the same expect block in this case
also I think just destroy is good enough since it detachs it before destroying it IIRC

same for the other one

Reply:
Makes sense, thanks!